File: task1/exponential.py
# -*- coding: utf-8 -*-
"""
Created on Mon Jun  2 20:07:28 2025

@author: DELL
"""

#exponential func
def fact(z):
    if z<=1:
        return 1
    else:
        return z* fact(z-1)
def exp(y):
    n=50
    exp=0
    for i in range (n+1):       
        m=(y**i)/fact(i)
        exp=m+exp
    return exp
test= exp(2)
#test
import math
test1=math.exp(2)
print(round(test,9)==round(test1,9))

File: task1/exponential.c
#include <stdio.h>
#include <math.h> 
int main() {
    double y;
    printf("Enter the value of y: ");
    scanf("%lf", &y);

    double result = exp(y);
    printf("e^%.2f ≈ %.10f\n", y, result);

    return 0;
}
double fact(int z) {
    if (z <= 1)
        return 1;
    else
        return z * fact(z - 1);
}

// Function to calculate exponential of y
double exp(double y) {
    int n = 50;
    double exp = 0.0;

    for (int i = 0; i <= n; i++) {        
        
        double m = pow(y, i) / fact(i);

        exp += m;
    }

    return exp;
}